pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -205,8 +205,9 @@ public DataTable processQuery(ServerQueryRequest queryRequest, ExecutorService e
         metadata.put(DataTable.NUM_SEGMENTS_MATCHED, "0");
       } else {
         TimerContext.Timer planBuildTimer = timerContext.startNewPhaseTimer(ServerQueryPhase.BUILD_QUERY_PLAN);
-        Plan globalQueryPlan =
-            _planMaker.makeInterSegmentPlan(segmentDataManagers, brokerRequest, executorService, remainingTimeMs);
+        Plan globalQueryPlan = _planMaker
+            .makeInterSegmentPlan(segmentDataManagers, queryContext.getBrokerRequest(), executorService,
+                remainingTimeMs);
```

Comment:
Is this because of step by step migration to QueryContext? We have already converted BrokerRequest to QueryContext at this point but for code that hasn't yet switched to QueryContext, a backward reference to broker request is still maintained inside QueryRequest to help with components that will still use BrokerRequest. Is this correct?

Reply:
@siddharthteotia Yes. I'll replace BrokerRequest with QueryContext step by step so that each PR won't be too big. Replacing BrokerRequest with QueryContext in `PlanNode` will be next step.

Reply:
Got it. Thanks.